packages/babel-parser/test/fixtures/experimental/pipeline-operator/proposal-fsharp-await/output.json

```diff
@@ -0,0 +1,181 @@
+              "line": 1,
+              "column": 15
+            },
+            "end": {
+              "line": 1,
+              "column": 19
+            },
+            "identifierName": "test"
+          },
+          "name": "test"
+        },
+        "generator": false,
+        "async": true,
+        "params": [],
+        "body": {
+          "type": "BlockStatement",
+          "start": 23,
+          "end": 47,
+          "loc": {
+            "start": {
+              "line": 1,
+              "column": 23
+            },
+            "end": {
+              "line": 3,
+              "column": 1
+            }
+          },
+          "body": [
+            {
+              "type": "ReturnStatement",
+              "start": 27,
+              "end": 45,
+              "loc": {
+                "start": {
+                  "line": 2,
+                  "column": 2
+                },
+                "end": {
+                  "line": 2,
+                  "column": 20
+                }
+              },
+              "argument": {
+                "type": "BinaryExpression",
+                "start": 34,
+                "end": 44,
+                "loc": {
+                  "start": {
+                    "line": 2,
+                    "column": 9
+                  },
+                  "end": {
+                    "line": 2,
+                    "column": 19
+                  }
+                },
+                "left": {
+                  "type": "PipelineHead",
+                  "start": 36,
+                  "end": 35,
+                  "loc": {
+                    "start": {
+                      "line": 2,
+                      "column": 11
+                    },
+                    "end": {
+                      "line": 2,
+                      "column": 10
+                    }
+                  },
+                  "head": {
+                    "type": "Identifier",
+                    "start": 34,
+                    "end": 35,
+                    "loc": {
+                      "start": {
+                        "line": 2,
+                        "column": 9
+                      },
+                      "end": {
+                        "line": 2,
+                        "column": 10
+                      },
+                      "identifierName": "x"
+                    },
+                    "name": "x"
+                  }
+                },
+                "operator": "|>",
+                "right": {
+                  "type": "PipelineBody",
+                  "start": 39,
+                  "end": 44,
+                  "loc": {
+                    "start": {
+                      "line": 2,
+                      "column": 14
+                    },
+                    "end": {
+                      "line": 2,
+                      "column": 19
+                    }
+                  },
+                  "body": {
+                    "type": "AwaitExpression",
```

Comment:
Maybe we could merge `PipelineBody` + `AwaitExpressoin` = `PipelineAwait`? So that plugins expecting `await` to have an `argument` don't break, and also because it's the only case where you can find `await` alone.

Reply:
This is a new AST node, correct? I don't have a problem with it, but I think it would alter the spec.

Reply:
According to the type for AwaitExpression, `argument` is already optional:

```js
export type AwaitExpression = NodeBase & {
  type: "AwaitExpression",
  argument: ?Expression,
};
```